projects/RabbitMQ.Client/client/framing/BasicAck.cs

```diff
@@ -40,35 +42,27 @@ internal sealed class BasicAck : Client.Impl.MethodBase
         public ulong _deliveryTag;
         public bool _multiple;
 
-        public BasicAck()
-        {
-        }
-
         public BasicAck(ulong DeliveryTag, bool Multiple)
         {
             _deliveryTag = DeliveryTag;
             _multiple = Multiple;
         }
 
-        public BasicAck(ReadOnlySpan<byte> span)
-        {
-            int offset = WireFormatting.ReadLonglong(span, out _deliveryTag);
-            WireFormatting.ReadBits(span.Slice(offset), out _multiple);
-        }
+        public BasicAck(ReadOnlySpan<byte> span) : this(NetworkOrderDeserializer.ReadUInt64(span, 0), (span[8] & 1) > 0) { }
 
         public override ProtocolCommandId ProtocolCommandId => ProtocolCommandId.BasicAck;
         public override string ProtocolMethodName => "basic.ack";
         public override bool HasContent => false;
 
+        [MethodImpl(RabbitMQMethodImplOptions.Optimized)]
         public override int WriteArgumentsTo(Span<byte> span)
         {
-            int offset = WireFormatting.WriteLonglong(span, _deliveryTag);
-            return offset + WireFormatting.WriteBits(span.Slice(offset), _multiple);
+            NetworkOrderSerializer.WriteUInt64(ref span[0], _deliveryTag);
+            span[8] = (byte)(_multiple ? 1 : 0);
+            return 9;
         }
 
-        public override int GetRequiredBufferSize()
-        {
-            return 8 + 1;
-        }
+        [MethodImpl(RabbitMQMethodImplOptions.Optimized)]
```

Comment:
This is also a good example. What should be flagged with the attribute what not? Also shouldn't the compiler inline this automatically due to the sealed type and the small size?
=> It will for the struct case over at #962

Reply:
I found that it didn't actually inline this when testing. I'll split this into smaller chunks for a more manageable review process. Overall this PR made some pretty hefty improvements to CPU usage though (due to the removed bounds-checks mostly).

Stay tuned.